### PR TITLE
opssight registry conf and image-registry support

### PR DESCRIPTION
**Pull request by msenmurugan**
opssight registry conf and image-registry support

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/738#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.8.x/OWNERS)~~ [msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/738#pullrequestreview-273654277" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/738#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.8.x/OWNERS)~~ [jdartigalongue,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/alert/v1/types.go [hunk 1]
```diff
@@ -60,9 +60,9 @@ type AlertSpec struct {
 	AlertMemory string `json:"alertMemory,omitempty"`
 	CfsslMemory string `json:"cfsslMemory,omitempty"`
 
-	DesiredState          string                    `json:"desiredState,omitempty"`
-	ImageRegistries       []string                  `json:"imageRegistries,omitempty"`
-	RegistryConfiguration api.RegistryConfiguration `json:"registryConfiguration,omitempty"`
+	DesiredState          string                     `json:"desiredState,omitempty"`
+	ImageRegistries       []string                   `json:"imageRegistries,omitempty"`
+	RegistryConfiguration *api.RegistryConfiguration `json:"registryConfiguration,omitempty"`
```
Comment (jdartigalongue) [inline pkg/api/alert/v1/types.go]: If we change RegistryConfiguration to a pointer in the Alert Spec then we need to check if this field is nil inside ctl_alert.go  before assigning registry / namespace / pull secrets.

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/738#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.8.x/OWNERS)~~ [msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/opssight/ctl_opssight.go [hunk 2]
```diff
@@ -45,57 +46,31 @@ import (
 // generate the spec by telling CRSpecBuilderFromCobraFlags what flags were changed.
 type CRSpecBuilderFromCobraFlags struct {
 	opsSightSpec                                    *opssightapi.OpsSightSpec
-	PerceptorName                                   string
-	PerceptorImage                                  string
-	PerceptorPort                                   int
+	IsUpstream                                      string
```
Comment (rrati) [inline pkg/opssight/ctl_opssight.go]: This seems to be used strictly for the component names and not the image locations.  What's the intent of this option?

File: pkg/opssight/prometheus.go [hunk 3]
```diff
@@ -139,8 +139,8 @@ func (p *SpecConfig) PerceptorMetricsService() (*components.Service, error) {
 	})
 
 	err := service.AddPort(horizonapi.ServicePortConfig{
-		Port:       9090,
-		TargetPort: "9090",
+		Port:       3006,
```
Comment (rrati) [inline pkg/opssight/prometheus.go]: Why is this changing from 9090 to 3006?

File: pkg/api/alert/v1/types.go (hunk 1 above)
Comment (msenmurugan) [reply pkg/api/alert/v1/types.go]: I think that should be fine right..

File: pkg/opssight/ctl_opssight.go [hunk 4]
```diff
@@ -151,6 +127,7 @@ func (ctl *CRSpecBuilderFromCobraFlags) SetPredefinedCRSpec(specType string) err
 		ctl.opsSightSpec = crddefaults.GetOpsSightUpstream()
 		ctl.opsSightSpec.Perceiver.EnablePodPerceiver = true
 		ctl.opsSightSpec.EnableMetrics = true
+		ctl.opsSightSpec.IsUpstream = true
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/opssight/ctl_opssight.go]: This method is to install the upstream version

**Comment by operator-r2d2** [comment]
@msenmurugan: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: pkg/opssight/ctl_opssight.go (hunk 2 above)
Comment (msenmurugan) [reply pkg/opssight/ctl_opssight.go]: both images and names

File: pkg/opssight/prometheus.go (hunk 3 above)
Comment (msenmurugan) [reply pkg/opssight/prometheus.go]: Someplace it is 3006 and in other place it is 9090. so making it consistent

File: pkg/api/opssight/v1/types.go [hunk 5]
```diff
@@ -95,32 +91,21 @@ type ScannerPod struct {
...
-	Name               string          `json:"name"`
-	Image              string          `json:"image"`
-	Port               int             `json:"port"`
 	InternalRegistries []*RegistryAuth `json:"internalRegistries"`
 	ImagePullerType    string          `json:"imagePullerType"`
-	ServiceAccount     string          `json:"serviceAccount"`
 }
 
 // ImagePerceiver stores the Image Perceiver configuration
 type ImagePerceiver struct {
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/api/opssight/v1/types.go]: removed

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/738#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/738#pullrequestreview-273778051" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.8.x/OWNERS)~~ [msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**